repo_commit.go

```diff
@@ -108,6 +108,31 @@ func (repo *Repository) getCommitIdOfPackedRef(refpath string) ([]byte, error) {
 	return nil, RefNotFound(refpath)
 }
 
+type packedRef struct {
+	commit  string
+	refpath string
+}
+
+// parseRefLine parses a line in the packed-refs file. This file
+// contains lines of the form `${commit-id} ${ref-name}`,
+// `^${commit-id}, and comment lines beginning with "#". This function
+// returns the parsed ref in the first case and an error in all other
+// cases.
+func parseRefLine(line string) (packedRef, error) {
+	parseErr := fmt.Errorf("could not parse ref from line %q", line)
```

Comment:
Style-wise, this should be named `errParse`. `parseErr` implies it's an error you got as a result of running a `parse` func, like `parseResult, parseErr := parse(foo, bar)`. `ErrFooBar` is the common style of `error` variable names.